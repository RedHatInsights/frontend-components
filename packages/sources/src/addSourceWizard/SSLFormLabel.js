import React from 'react';
import { QuestionCircleIcon } from '@patternfly/react-icons';
import { Popover, Text, TextContent, TextVariants } from '@patternfly/react-core';

const SSLFormLabel = () => (
    <React.Fragment>
        SSL Certificate&nbsp;
        <Popover
            aria-label="Help text"
            maxWidth="50%"
            bodyContent={
                <TextContent>
                    <Text component={ TextVariants.p }>
                            You can obtain your OpenShift Container Platform providerâ€™s CA
                            certificate for all endpoints (default, metrics, alerts) from
                        <b>/etc/origin/master/ca.crt</b>.
                    </Text>
                    <Text component={ TextVariants.p }>
                            Paste the output (a block of text starting with --BEGIN CERTIFICATE--)
                            into the Trusted CA Certificates field.
                    </Text>
                </TextContent>
            }
        >
            <QuestionCircleIcon />
        </Popover>
    </React.Fragment>
);

export default SSLFormLabel;
