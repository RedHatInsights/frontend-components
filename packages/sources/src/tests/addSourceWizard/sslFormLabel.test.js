import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import SSLFormLabel from '../../addSourceWizard/SSLFormLabel';

describe('SSLFormLabel', () => {
    it('renders loading step correctly', () => {
        const wrapper = shallow(<SSLFormLabel />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
