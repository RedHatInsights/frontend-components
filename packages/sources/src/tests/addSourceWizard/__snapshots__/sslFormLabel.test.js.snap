// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SSLFormLabel renders loading step correctly 1`] = `
<Fragment>
  SSL Certificate 
  <span
    onClick={[Function]}
  >
    <Popover
      appendTo={[Function]}
      aria-label="Help text"
      bodyContent={
        <TextContent>
          <Text
            component="p"
          >
            You can obtain your OpenShift Container Platform provider’s CA certificate for all endpoints (default, metrics, alerts) from
            <b>
              /etc/origin/master/ca.crt
            </b>
            .
          </Text>
          <Text
            component="p"
          >
            Paste the output (a block of text starting with --BEGIN CERTIFICATE--) into the Trusted CA Certificates field.
          </Text>
        </TextContent>
      }
      boundary="window"
      className=""
      closeBtnAriaLabel="Close"
      distance={25}
      enableFlip={true}
      flipBehavior={
        Array [
          "top",
          "right",
          "bottom",
          "left",
          "top",
          "right",
          "bottom",
        ]
      }
      footerContent={null}
      headerContent={null}
      hideOnOutsideClick={true}
      isVisible={null}
      maxWidth="50%"
      onHidden={[Function]}
      onHide={[Function]}
      onMount={[Function]}
      onShow={[Function]}
      onShown={[Function]}
      position="top"
      shouldClose={[Function]}
      zIndex={9999}
    >
      <QuestionCircleIcon
        color="currentColor"
        noVerticalAlign={false}
        size="sm"
        title={null}
      />
    </Popover>
  </span>
</Fragment>
`;
